### ISSUE TITLE
Minijinja test failures

### ISSUE THREAD

**Issue by psibi** [mitsuhiko/minijinja#402]
## Description

 Tried to package minijinja-cli for nixpkgs and faced test failures when executing the test suite there.

## Reproduction steps

1. Use nix to build minijinja-cli. I use this command:

```
nix-build -E '(builtins.getFlake "nixpkgs").legacyPackages.x86_64-linux.callPackage ./default.nix {}'
```

I can share the default.nix if you would want, but I don't think the issue is nix specific unless I'm missing something.

2. See test failures

Attached the full failure logs here: https://gist.github.com/psibi/ad9b35eb8019a92899ca68ccd59634d5

Additional helpful information:

- Version of minijinja: 1.0.12
- Version of rustc: 1.75.0
- Operating system and version:

```
❯ uname -a
Linux arya 6.1.63 #1-NixOS SMP PREEMPT_DYNAMIC Mon Nov 20 10:52:19 UTC 2023 x86_64 GNU/Linux
```

Note that nix package manager downloads the tarball from github release tag and then tries to build the package.

## What did you expect

No test failures.

**Comment by mitsuhiko** [comment]
Please share how you are invoking the tests. The tests need a specific set of features invoked which is why `make test` passes those explicitly. If you are just running the tests with `cargo test` they are likely to fail.

**Comment by psibi** [comment]
The tests are invoked automatically by the above `nix` command that I have shared. By default, the nix package manager just runs `cargo test`. 

I just looked at the makefile and I see what you mean.  Also I see the following line as part of tests:

```
rustup component add rustfmt 2> /dev/null
```

The above step will fail in a nix test environment as it is sand-boxed and doesn't allow any network connection. I will have look more into the Makefile to figure out the details, thanks for the pointer.
